dataservice/api/biospecimen/models.py

```diff
@@ -69,6 +68,10 @@ class Biospecimen(db.Model, Base):
                                db.ForeignKey('participant.kf_id'),
                                nullable=False,
                                doc='The kf_id of the biospecimen\'s donor')
+    sequencing_center_id = db.Column(KfId(),
+                                     db.ForeignKey('sequencing_center.kf_id'),
+                                     nullable=False,
```

Comment:
Ya i also thought about it like in the cases of where the collected biospecimens are not sent for sequencing etc. in that case we need to remove nullable=False here. Only thing that concerns me is; we usually have sequencing_center information when creating biospecimen and sequencing experiment . Making foreign key nullable in 1:n relationship makes difficult to join if multiple null values exists in the sequencing_center table. For suppose, Biospecimen1 from one center has null in sequencing_center_id and and also Biospecimen2 from another center has null value. Now we get in to situation of dealing with foreign key issues.

Reply:
I'm not sure I understand... there should never be null values in the sequencing center table. The null values would be foreign keys to the sequencing center table from biospecimen or sequencing experiment. Also i don't think there would be issues with joins between sequencing center and biospecimen/sequencing experiment where the foreign keys are null. We would just need to make sure the join is not an inner join so that we still get back biospecimens/sequencing experiments from the join result.

Reply:
`nullable=False` seems to make sense to me. AFAIK, we receive initial information about the sample/biospecimen from the shipment manifests, which are manifests provided to a specific sequencing center.

Reply:
I think @parimalak has a good point though, sometimes biospecimens will exist and we won't have sequencing center info for them.

Reply:
Would it make sense to make `nullable=False` now since all of the current use cases there is a sequencing center (as the info is coming from a shipping manifest) and relax it later if the case of biospecimens without an assigned center does come up?